README.md
# Fan-Car-Pet_Class
This repository consists Fan, Car and Pet Class. Each class has its own factors and functions which significantly highlights data manipulation of information inside a class, outside the class, and within the sub-class.

## Fan Class
(The Fan class).  The class named Fan to represent a fan. The class contains:

■ Three constants named SLOW, MEDIUM, and FAST with the values 1, 2, and 3 to denote the fan speed.

■ A private int data field named speed that specifies the speed of the fan.

■ A private bool data field named on that specifies whether the fan is on (the default is False).

■ A private float data field named radius that specifies the radius of the fan.

■ A private string data field named color that specifies the color of the fan.

■ The accessor(getters)  and mutator(setters)  methods for all four data fields.

■ A constructor that creates a fan with the specified speed (default SLOW), radius (default 5), color (default blue), and on (default False).

Write a test program named TestFan that creates two Fan objects. For the first object, assign the maximum speed, radius 10, color yellow, and turn it on. Assign medium speed, radius 5, color blue, and turn it off for the second object. Display each object’s speed, radius, color, and on properties.

## Car Class
A class name Car that has the following data attributes:

• _ _year_model (for the car’s year model)

• _ _make (for the make of the car)

• _ _speed (for the car’s current speed)


The Car class have an _ _init_ _ method that accepts the car’s year model and make as arguments. These values assigns to the object’s _ _year_model and _ _make data attributes. It assigns 0 to the _ _speed data attribute as default.

The class have the following methods:

• accelerate()

The accelerate method add 5 to the speed data attribute each time it is called.

• brake()

The brake method subtract 5 from the speed data attribute each time it is called.

• get_speed()

The get_speed method return the current speed.


Next, it is followed bya program named Car-Test-Driver that creates a Car object then calls the accelerate method five times. After each call to the accelerate method, get the current speed of the car and display it. Then call the brake method five times. After each call to the brake method, get the current speed of the car and display it.

## Pet Class
It is a class named Pet, which follows data attributes:

• _ _name (for the name of a pet)

• _ _animal_type (for the type of animal that a pet is. Example values are ‘Dog’, ‘Cat’, and ‘Bird’)

• _ _age (for the pet’s age)

The Pet class have an _ _init_ _ method that creates these attributes. It also have the following methods:

• set_name()

This method assigns a value to the _ _name field.

• set_animal_type()

This method assigns a value to the _ _animal_type field.

• set_age()

This method assigns a value to the _ _age field.

• get_name()

This method returns the value of the _ _ name field.

• get_animal_type()

This method returns the value of the _ _animal_type field.

• get_age()

This method returns the value of the _ _age field.


Once the class has written, a program that creates an object of the class and prompts the user to enter the name, type, and age of his or her pet. It is a program named Pet-Test-Driver.py. This data should be stored as the object’s attributes. Use the object’s accessor methods to retrieve the pet’s name, type, and age and display this data on the screen. The stored data goes to history.txt.




History.txt
Animal name: CLEAVELAND
Animal Type: WOLF
Age: 39

Animal name: MINATOZAKI SANA
Animal Type: DOG
Age: 55

Animal name: JEONGYEON JEON
Animal Type: OSTRICH
Age: 33



Car/Car.py
class Car:
    def __init__(self, year_model, make, speed = 0):
        self.__year_model = year_model
        self.__make = make
        self.__speed = speed
    
    def get_year_model(self):
        return self.__year_model
    
    def get_make(self):
        return self.__make
    
    def get_speed(self):
        return self.__speed
    
    def accelerate(self):
        self.__speed = int(self.__speed) + 5
        print("Accelerate to: ", self.__speed, "Kilometers per hour")
        return self.__speed
    
    def brake(self):
        self.__speed = int(self.__speed) - 5
        print("Brake to: ", self.__speed, "Kilometers per hour")
        return self.__speed
    
    def show(self):
        print("Year Model: ", self.__year_model)
        print("Make: ", self.__make)
        print("Current Speed: ", self.__speed, "Kilometers per second")
    



Car/Car-Test-Driver.py
from Car import Car
import pyfiglet
from colored import fg

#pyfiglet format :
# color = fg('blue')
#text_print = (f"Text")
#print(color + pyfiglet.figlet_format(text_print , font= "digital")) 
color = fg('blue')
#Colors
BLUE = '\033[94m'
DARKCYAN = '\033[36m'
PURPLE = '\033[95m'
GREEN = '\033[92m'
RED = '\033[91m'
YELLOW = '\033[93m'
BOLD = '\033[1m'
END = '\033[0m'
BORDER = 100*'='
print(YELLOW+BOLD+BORDER+END)
print(YELLOW+BOLD+BORDER+END)

print(GREEN+BOLD+"Inputs: "+END)

year = input("Insert year model: ")
make = input("Insert the body make: ")

print('\n')

print(YELLOW+BOLD+BORDER+END)
print(YELLOW+BOLD+BORDER+END)

car1 = Car(year, make)

print(BLUE+BOLD+"Status: "+END)
car1.show()

print('\n')

print(YELLOW+BOLD+BORDER+END)
print(YELLOW+BOLD+BORDER+END)
print(GREEN+BOLD+"Accelerate: "+END)

for x in range(5):
    car1.accelerate()

print('\n')

print(BOLD+BORDER+END)

print(BLUE+BOLD+"Status: "+END)
car1.show()

print('\n')

print(BOLD+BORDER+END)
print(BOLD+BORDER+END)

print(GREEN+BOLD+"Brake: "+END)

for x in range(5):
    car1.brake()

print('\n')

print(BOLD+BORDER+END)

print(BLUE+BOLD+"Status: "+END)
car1.show()

print('\n')

print(BOLD+BORDER+END)
print(BOLD+BORDER+END)

Fan/Fan-Test-Driver.py
from Fan import Fan
import pyfiglet
from colored import fg

#pyfiglet format :
# color = fg('blue')
#text_print = (f"Text")
#print(color + pyfiglet.figlet_format(text_print , font= "digital")) 
color = fg('blue')
#Colors
BLUE = '\033[94m'
DARKCYAN = '\033[36m'
PURPLE = '\033[95m'
GREEN = '\033[92m'
RED = '\033[91m'
YELLOW = '\033[93m'
BOLD = '\033[1m'
END = '\033[0m'
BORDER = 100*'='
print(GREEN+BOLD+BORDER+END)
print(GREEN+BOLD+BORDER+END)

#default:
f1 = Fan()

print('\n')
print(PURPLE+BOLD+"Fan Default: "+END)
print('\n')

f1.show()
print('\n')

print(GREEN+BOLD+BORDER+END)
print(GREEN+BOLD+BORDER+END)

print('\n')
print(BOLD+BORDER+END)
print(BOLD+BORDER+END)

#object 1
print('\n')

print(BLUE+BOLD+"Fan 1:"+END)

speed = input("How fast do you want your fan?('SLOW','MEDIUM', or 'FAST'): ")
radius = input("How wide is your fan blades?('5.0' to '10.0'): ")
color = input("Which color do you like?: ")
power = input("Do you want to turn it on?: ")

f2 = Fan()
print('\n')
print('\n')

f2.set_speed(speed)
f2.set_radius(radius)
f2.set_color(color)
f2.set_on(power)

print(RED+BOLD+"Fan 1:"+END)

print('\n')

print(BOLD+BORDER+END)
print(BOLD+BORDER+END)

print('\n')

print("Fan speed: ",f2._Fan__speed)
print("Fan blade Radius: ", f2._Fan__radius)
print("Fan Color: ", f2._Fan__color)
print("Power on?: ", f2._Fan__on)

print('\n')
print(BOLD+BORDER+END)
print(BOLD+BORDER+END)

print('\n')

print(PURPLE+BOLD+BORDER+END)
print(PURPLE+BOLD+BORDER+END)

#object 1
print('\n')

print(YELLOW+BOLD+"Fan 2:"+END)

speed = input("How fast do you want your fan?('SLOW','MEDIUM', or 'FAST'): ")
radius = input("How wide is your fan blades?('5.0' to '10.0'): ")
color = input("Which color do you like?: ")
power = input("Do you want to turn it on?: ")

f3 = Fan()

print('\n')
print('\n')

f3.set_speed(speed)
f3.set_radius(radius)
f3.set_color(color)
f3.set_on(power)

print(PURPLE+BOLD+BORDER+END)
print(PURPLE+BOLD+BORDER+END)

print('\n')

print(YELLOW+BOLD+"Fan 2:"+END)

print('\n')

print("Fan speed: ",f3._Fan__speed)
print("Fan blade Radius: ", f3._Fan__radius)
print("Fan Color: ", f3._Fan__color)
print("Power on?: ", f3._Fan__on)

print('\n')
print('\n')

print(PURPLE+BOLD+BORDER+END)
print(PURPLE+BOLD+BORDER+END)

Pet/Pet-Test-Driver.py
from Pet import Pet
import pyfiglet
from colored import fg

#pyfiglet format :
# color = fg('blue')
#text_print = (f"Text")
#print(color + pyfiglet.figlet_format(text_print , font= "digital")) 
color = fg('blue')
#Colors
BLUE = '\033[94m'
DARKCYAN = '\033[36m'
PURPLE = '\033[95m'
GREEN = '\033[92m'
RED = '\033[91m'
YELLOW = '\033[93m'
BOLD = '\033[1m'
END = '\033[0m'
BORDER = 100*'='

print(BOLD+BORDER+END)
print(BOLD+BORDER+END)

pet1 = Pet('Peter Griffin', 'Beluga Whale', 89)

print(PURPLE+BOLD+"Sample Inputs"+END)
pet1.show()

print(BOLD+BORDER+END)
print(BOLD+BORDER+END)

print('\n')

print(DARKCYAN+BOLD+BORDER+END)
print(DARKCYAN+BOLD+BORDER+END)

print(BLUE+BOLD+"Enter your pet's data: "+END)

name = input("Enter the name of your pet: ")
animal_type = input("Enter the animal type of your pet: ")
age = input("Enter the age of your pet in month: ")

pet1.set_name(name)
pet1.set_animal_type(animal_type)
pet1.set_age(age)


print(DARKCYAN+BOLD+BORDER+END)
print(DARKCYAN+BOLD+BORDER+END)

print('\n')

print(DARKCYAN+BOLD+BORDER+END)
print(DARKCYAN+BOLD+BORDER+END)

print(BLUE+BOLD+"Pet Information: "+END)
pet1.show()

history = open('History.txt', 'a')
history.write("Animal name: "),history.write(pet1.get_name()), history.write('\n')
history.write("Animal Type: "),history.write(pet1.get_animal_type()), history.write('\n')  
history.write("Age: "),history.write(str(pet1.get_age())), history.write('\n'), history.write('\n') 
history.close()

print('\n')

print(GREEN+BOLD+"Thank you for your cooperation."+END)
print(GREEN+BOLD+"Your information has been successfuly saved."+END)
print(DARKCYAN+BOLD+BORDER+END)
print(DARKCYAN+BOLD+BORDER+END)

Fan/Fan.py
class Fan:
    def __init__(self, speed = 'SLOW' , radius = 5.0, color = 'blue', on = 'yes'):
        speed = speed.upper()
        if speed == 'SLOW' or speed == 'MEDIUM' or speed == 'FAST':
            if speed == 'SLOW':
                self.__speed = int(1)
            elif speed == 'MEDIUM':
                self.__speed = int(2)
            elif speed == 'FAST':
                self.__speed = int(3)
                
        if float(radius) >= float(5.0) or float(radius) <= float(10.0):
            self.__radius = float(radius)

        self.__color = str(color.lower())
        on = on.lower()
        if on == 'yes' or on == 'no':
            if on == 'yes':
                self.__on = True
            if on == 'no':
                self.__on = False
        

    def get_speed(self):
        return self.__speed
    
    def set_speed(self, speed):
        speed = speed.upper()
        if speed == 'SLOW' or speed == 'MEDIUM' or speed == 'FAST':
            if speed == 'SLOW':
                self.__speed = int(1)
            elif speed == 'MEDIUM':
                self.__speed = int(2)
            elif speed == 'FAST':
                self.__speed = int(3)

        else:
            print("Please choose from 'SLOW', 'MEDIUM', and 'FAST' ONLY.")
    
    def get_radius(self):
        return self.__radius
    
    def set_radius(self, radius):
        if float(radius) >= float(5.0) or float(radius) <= float(10.0):
            self.__radius = float(radius)
        
        else:
            print("Please choose from '5.0' to '10.0' ONLY.")

    def get_on(self):
        return self.__on
    
    def set_on(self, on):
        on = on.lower()
        if on == 'yes' or on == 'no':
            if on == 'yes':
                self.__on = True
            if on == 'no':
                self.__on = False
            
        else:
            print("Please choose from 'yes' or 'no' ONLY.")
    
    def get_color(self):
        return self.__color
    
    def set_color(self, color):
        self.__color = color.lower()

    def show(self):
        print("Speed: ", self.get_speed())
        print("Radius: ", self.get_radius())
        print("Is it on?: ", self.get_on())
        print("Color: ", self.get_color())
    
